compiler/sem/pragmas.nim

```diff
@@ -1563,90 +1377,56 @@ proc prepareSinglePragma(
-        elif sym.typ != nil:
-          incl(sym.typ.flags, tfNeedsFullInit)
         else:
-          result = invalidPragma(c, it)
+          assert sym.typ != nil
+          incl(sym.typ.flags, tfNeedsFullInit)
       of wByRef:
         result = noVal(c, it)
-        if sym == nil or sym.typ == nil:
-          var tmp = c.config.options
-          result = processOption(c, it, tmp)
-          c.config.options = tmp
-        else:
-          incl(sym.typ.flags, tfByRef)
+        assert sym != nil and sym.typ != nil
+        incl(sym.typ.flags, tfByRef)
       of wByCopy:
         result = noVal(c, it)
-        if sym.kind != skType or sym.typ == nil:
-          result = invalidPragma(c, it)
-        else:
-          incl(sym.typ.flags, tfByCopy)
+        assert sym.kind == skType and sym.typ != nil
+        incl(sym.typ.flags, tfByCopy)
       of wInject, wGensym:
         # We check for errors, but do nothing with these pragmas otherwise
         # as they are handled directly in 'evalTemplate'.
         result = noVal(c, it)
-        if sym == nil: result = invalidPragma(c, it)
-      of wLine:
-        result = pragmaLine(c, it)
       of wRaises, wTags:
         result = pragmaRaisesOrTags(c, it)
       of wLocks:
-        if sym == nil:
-          result = pragmaLockStmt(c, it)
-        elif sym.typ == nil:
-          result = invalidPragma(c, it)
-        else:
           (sym.typ.lockLevel, result) = pragmaLocks(c, it)
           if result.isNil:
             result = it
       of wBitsize:
-        if sym == nil or sym.kind != skField:
-          result = invalidPragma(c, it)
-        else:
+          assert sym.kind == skField
           (sym.bitsize, result) = intLitToIntOrErr(c, it)
           if result.isNil: result = it
           if sym.bitsize <= 0:
             result = c.config.newError(it,
                       PAstDiag(kind: adSemBitsizeRequiresPositive))
       of wGuard:
-        result = it
-        if sym == nil or sym.kind notin {skVar, skLet, skField}:
-          result = invalidPragma(c, it)
-        else:
+          assert sym.kind in {skVar, skLet, skField}
           sym.guard = pragmaGuard(c, it, sym.kind)
           if sym.guard != nil and sym.guard.kind == skError and sym.guard.ast != nil and sym.guard.ast.kind == nkError:
             result = sym.guard.ast
           else:
             result = it
       of wGoto:
-        result = it
-        if sym == nil or sym.kind notin {skVar, skLet}:
-          result = invalidPragma(c, it)
-        else:
-          sym.flags.incl sfGoto
+        result = noVal(c, it)
+        assert sym.kind in {skVar, skLet}
+        sym.flags.incl sfGoto
       of wExportNims:
-        result = it
-        if sym == nil:
-          result = invalidPragma(c, it)
-        else:
+          # XXX: modifying the module graph during application of a symbol
+          #      operator doesn't seem like a good idea...
           result = magicsys.registerNimScriptSymbol2(c.graph, sym)
```

Comment:
This is clean-up I forgot, we can drop the old `registerNimScriptSymbol` and take over the name. Doesn't have to be tackled in this PR, I can do it after.